### ISSUE TITLE
Wrong type in class field with __future__ annotations, when class has method with the same name

### ISSUE THREAD

**Issue by DanilaMihailov** [astral-sh/ty#1999]
### Summary

python 3.9.21

If class has method with the same name as builtin type, it gets used for types. 

**Only** when `from __future__ import annotations` used.

```python
from __future__ import annotations


class MyClass:
    field: set[str]

    def set(self): ... # named the same as builtin type
```

Error:
```
error[invalid-type-form]: Invalid subscript of object of type `def set(self) -> Unknown` in type expression
 --> ty_repro.py:5:12
  |
4 | class MyClass:
5 |     field: set[str]
  |            ^^^^^^^^
6 |
7 |     def set(self): ...
  |
info: rule `invalid-type-form` is enabled by default
```

### Version

ty 0.0.2 (42835578d 2025-12-16)

**Comment by AlexWaygood** [comment]
Thanks for the report! Please see https://github.com/astral-sh/ty/issues/1747
